test/gtest/uct/test_peer_failure.cc

```diff
@@ -612,3 +614,91 @@ UCS_TEST_SKIP_COND_P(test_uct_peer_failure_keepalive, killed,
 
 UCT_INSTANTIATE_NO_SELF_TEST_CASE(test_uct_peer_failure_keepalive)
 _UCT_INSTANTIATE_TEST_CASE(test_uct_peer_failure_keepalive, cuda_ipc);
+
+
+class test_ucp_peer_failure_rma_zcopy : public test_uct_peer_failure
+{
+public:
+    test_ucp_peer_failure_rma_zcopy()
+    {
+        m_dummy_comp.func   = NULL;
+        m_dummy_comp.count  = INT_MAX;
+        m_dummy_comp.status = UCS_OK;
+    }
+
+    // Pretend peer failure by using invalid address for cma and invalid key
+    // for knem
+    uint64_t failed_peer_address(mapped_buffer *recv_buffer)
+    {
+        if (has_cma()) {
+            return  0xDEADBEAF;
+        } else {
+            EXPECT_TRUE(has_knem());
+            return recv_buffer->addr();
+        }
+    }
+
+    uct_rkey_t failed_peer_rkey(mapped_buffer *recv_buffer)
+    {
+        if (has_cma()) {
+            return 0ul;
+        } else {
+            EXPECT_TRUE(has_knem());
+            uct_knem_key_t *rk = (uct_knem_key_t *)recv_buffer->rkey();
+            m_knem_key         = *rk;
+            m_knem_key.cookie = 0;
+            return reinterpret_cast<uct_rkey_t>(&m_knem_key);
+        }
+    }
+
+    uct_completion_t m_dummy_comp;
+    uct_knem_key_t   m_knem_key;
+};
+
+UCS_TEST_SKIP_COND_P(test_ucp_peer_failure_rma_zcopy, put,
+                     !check_caps(UCT_IFACE_FLAG_ERRHANDLE_PEER_FAILURE |
+                                 UCT_IFACE_FLAG_PUT_ZCOPY))
+{
+    {
+        scoped_log_handler slh(wrap_errors_logger);
+        const size_t size =128;
+        mapped_buffer sendbuf(size, 0, *m_sender);
+        mapped_buffer recvbuf(size, 0, *m_receivers.front());
+
+        ucs_status_t status = uct_ep_put_zcopy(m_sender->ep(0), sendbuf.iov(), 1,
```

Comment:
can put/get flows be tested as variants?

Reply:
could be done this way, but it seems to be simple enough now, so I'd leave it as it is

Reply:
also variants for UCT tests were recently removed, so could not do that anyway

Reply:
I see, thanks
probably, it's worth wrapping them by to a common wrapper which will take `uct_ep_put_zcopy`/`uct_ep_get_zcopy` as an argument and execute the operation, wdyt?